internal/command/views/test.go

```diff
@@ -732,6 +736,276 @@ func (t *TestJSON) TFCRetryHook(attemptNum int, resp *http.Response) {
+type TestJUnitXMLFile struct {
+	filename string
+	err      error
+}
+
+var _ Test = (*TestJUnitXMLFile)(nil)
+
+// NewTestJUnitXML returns a [Test] implementation that will, when asked to
+// report "conclusion", write a JUnit XML report to the given filename.
+//
+// If the file already exists then this view will silently overwrite it at the
+// point of being asked to write a conclusion. Otherwise it will create the
+// file at that time. If creating or overwriting the file fails, a subsequent
+// call to method Err will return information about the problem.
+func NewTestJUnitXMLFile(filename string) *TestJUnitXMLFile {
+	return &TestJUnitXMLFile{
+		filename: filename,
+	}
+}
+
+// Err returns an error that the receiver previously encountered when trying
+// to handle the Conclusion event by creating and writing into a file.
+//
+// Returns nil if either there was no error or if this object hasn't yet been
+// asked to report a conclusion.
+func (v *TestJUnitXMLFile) Err() error {
+	return v.err
+}
+
+func (v *TestJUnitXMLFile) Abstract(suite *moduletest.Suite) {}
+
+func (v *TestJUnitXMLFile) Conclusion(suite *moduletest.Suite) {
+	xmlSrc, err := junitXMLTestReport(suite)
+	if err != nil {
+		v.err = err
+		return
+	}
+	err = os.WriteFile(v.filename, xmlSrc, 0660)
+	if err != nil {
+		v.err = err
+		return
+	}
+}
+
+func (v *TestJUnitXMLFile) File(file *moduletest.File, progress moduletest.Progress) {}
+
+func (v *TestJUnitXMLFile) Run(run *moduletest.Run, file *moduletest.File, progress moduletest.Progress, elapsed int64) {
+}
+
+func (v *TestJUnitXMLFile) DestroySummary(diags tfdiags.Diagnostics, run *moduletest.Run, file *moduletest.File, state *states.State) {
+}
+
+func (v *TestJUnitXMLFile) Diagnostics(run *moduletest.Run, file *moduletest.File, diags tfdiags.Diagnostics) {
+}
+
+func (v *TestJUnitXMLFile) Interrupted() {}
+
+func (v *TestJUnitXMLFile) FatalInterrupt() {}
+
+func (v *TestJUnitXMLFile) FatalInterruptSummary(run *moduletest.Run, file *moduletest.File, states map[*moduletest.Run]*states.State, created []*plans.ResourceInstanceChangeSrc) {
+}
+
+func (v *TestJUnitXMLFile) TFCStatusUpdate(status tfe.TestRunStatus, elapsed time.Duration) {}
+
+func (v *TestJUnitXMLFile) TFCRetryHook(attemptNum int, resp *http.Response) {}
+
+func junitXMLTestReport(suite *moduletest.Suite) ([]byte, error) {
+	var buf bytes.Buffer
+	enc := xml.NewEncoder(&buf)
+	enc.EncodeToken(xml.ProcInst{
+		Target: "xml",
+		Inst:   []byte(`version="1.0" encoding="UTF-8`),
+	})
+	enc.Indent("", "  ")
+
+	// Some common element/attribute names we'll use repeatedly below.
+	suitesName := xml.Name{Local: "testsuites"}
+	suiteName := xml.Name{Local: "testsuite"}
+	caseName := xml.Name{Local: "testcase"}
+	nameName := xml.Name{Local: "name"}
+	testsName := xml.Name{Local: "tests"}
+	failuresName := xml.Name{Local: "failures"}
+	errorsName := xml.Name{Local: "errors"}
+
+	enc.EncodeToken(xml.StartElement{Name: suitesName})
+	for _, file := range suite.Files {
```

Comment:
The test file itself can have error diagnostics attached to it, and I think they're not being exposed anywhere at the moment. Is there somewhere we can surface those?